lasertagger/run_lasertagger_utils.py

```diff
@@ -323,3 +350,33 @@ def _get_decoder_params(config, use_tpu,
       # tags to equal the number of source tokens.
       extra_decode_length=0)
   return params
+
+
+def _update_loss_with_weight(loss, weight, filter_labels, labels):
+    """ Returns loss adjusted with weights.
+    
+    Args:
+      loss: original loss before weighting
+      weight: weight for this edit
+      filter_labels: the id number of the vocab corresponding to the edit
+      labels: predicted labels
+    
+    Returns:
+      Updated loss
+    """
+    if weight == 1:
+        return loss
+    else:
+        filter_label_mask = tf.constant(0.0, dtype="float32", shape=labels.get_shape())
+        for filter_label in filter_labels:
+            filter_label_mask = tf.math.add(
+                tf.cast(tf.math.equal(tf.constant(filter_label, dtype="int32"), labels), tf.float32), 
+                filter_label_mask)
+        loss_weight = tf.math.scalar_mul(tf.constant(weight - 1, dtype="float32"), filter_label_mask)
```

Comment:
why weight - 1 ?

Reply:
I add that 1 back on line 378-379. The idea is that for all the elements in the loss tensor whose corresponding element is 0 in the filter_label_mask, I keep that element the same. For all elements in the loss tensor whose corresponding element in the filter_label_mask is 1, I multiply that element by the weight. So basically I need a tensor where all elements are 1 except a few that need to be weighted by the weight. And line 374-375 along with line 378-379 is constructing such tensor.

Reply:
Got it, thanks for the explanation.